src/wallet/components/Wallet.tsx

```diff
@@ -1,45 +1,45 @@
-import { Children, useMemo, useRef } from 'react';
+import { useEffect } from 'react';
 import { useIsMounted } from '../../core-react/internal/hooks/useIsMounted';
 import { useTheme } from '../../core-react/internal/hooks/useTheme';
-import { findComponent } from '../../core-react/internal/utils/findComponent';
 import { cn } from '../../styles/theme';
-import { useOutsideClick } from '../../ui/react/internal/hooks/useOutsideClick';
 import type { WalletReact } from '../types';
-import { ConnectWallet } from './ConnectWallet';
-import { WalletDropdown } from './WalletDropdown';
 import { WalletProvider, useWalletContext } from './WalletProvider';
 
-const WalletContent = ({ children, className }: WalletReact) => {
-  const { isOpen, setIsOpen } = useWalletContext();
-  const walletContainerRef = useRef<HTMLDivElement>(null);
-
-  useOutsideClick(walletContainerRef, () => {
-    if (isOpen) {
-      setIsOpen(false);
-    }
-  });
+function WalletContent({ children, className }: WalletReact) {
+  const componentTheme = useTheme();
+  const {
+    isOpen,
+    handleClose,
+    containerRef: walletContainerRef,
+  } = useWalletContext();
 
-  const { connect, dropdown } = useMemo(() => {
-    const childrenArray = Children.toArray(children);
-    return {
-      connect: childrenArray.find(findComponent(ConnectWallet)),
-      dropdown: childrenArray.find(findComponent(WalletDropdown)),
+  useEffect(() => {
+    const handleClickOutsideComponent = (event: MouseEvent) => {
+      if (
+        walletContainerRef?.current &&
+        !walletContainerRef?.current.contains(event.target as Node) &&
+        isOpen
+      ) {
+        handleClose();
+      }
     };
-  }, [children]);
+
+    document.addEventListener('click', handleClickOutsideComponent);
+    return () =>
+      document.removeEventListener('click', handleClickOutsideComponent);
+  }, [isOpen, handleClose, walletContainerRef]);
```

Comment:
Let's use the useClickOutside hook!

Reply:
oh cool, i think you added this after i made my refactor, super helpful, thanks!